src/System.Management.Automation/engine/hostifaces/LocalPipeline.cs

```diff
@@ -152,13 +155,30 @@ protected override void StartPipelineExecution()
 
             PSThreadOptions memberOptions = this.IsNested ? PSThreadOptions.UseCurrentThread : this.LocalRunspace.ThreadOptions;
 
+            // If impersonation identity flow is requested, then get current thread impersonation, if any.
+#if !UNIX
+            ThreadStart invokeThreadProcDelegate;
+            if ((InvocationSettings != null) && InvocationSettings.FlowImpersonationPolicy &&
+                Utils.TryGetWindowsImpersonatedIdentity(out _identityToImpersonate))
```

Comment:
Question: I see the `identityToImpersonate` is disposed at some places and not at some other places. For example, this instance is never disposed after use. Shall we always dispose it after use?

Reply:
Yes, since WindowsIdentity has a Dispose() method I assume it has system resources to release.  So I call it whenever I feel I can do it safely to release system resources as soon as possible.  For other cases, like LocalPipeline, I am not sure when it is safe to dispose and I let finalizer thread do it ... I don't want to get an "invalid handle" exception.

For WSManTransportManger, I worry about multiple calls to the Dipose() method causing a handle exception, so I again don't dispose the WindowsIdentity object.  But this should be Ok since it is only used at the end of a remote session and finalizer should do the job.